apps/emqx_gcp_device/src/emqx_gcp_device_api.erl

```diff
@@ -0,0 +1,488 @@
+                }
+            )},
+        {location,
+            hoconsc:mk(
+                binary(),
+                #{
+                    desc => ?DESC(location),
+                    default => <<>>,
+                    example => <<"europe-west1">>
+                }
+            )},
+        {keys,
+            hoconsc:mk(
+                ?ARRAY(hoconsc:ref(key)),
+                #{
+                    desc => ?DESC(keys),
+                    default => []
+                }
+            )},
+        {config,
+            hoconsc:mk(
+                binary(),
+                #{
+                    desc => ?DESC(config),
+                    required => true,
+                    example => <<"bXktY29uZmln">>
+                }
+            )}
+    ];
+fields(gcp_device_info) ->
+    [
+        {deviceid,
+            hoconsc:mk(
+                binary(),
+                #{
+                    desc => ?DESC(deviceid),
+                    required => true,
+                    example => <<"c2-ec-x509">>
+                }
+            )}
+    ] ++ fields(gcp_device);
+fields(gcp_device_all_info) ->
+    [
+        {created_at,
+            hoconsc:mk(
+                non_neg_integer(),
+                #{
+                    desc => ?DESC(created_at),
+                    required => true,
+                    example => 1690484400
+                }
+            )}
+    ] ++ fields(gcp_device_info);
+fields(gcp_exported_device) ->
+    [
+        {deviceid,
+            hoconsc:mk(
+                binary(),
+                #{
+                    desc => ?DESC(deviceid),
+                    required => true,
+                    example => <<"c2-ec-x509">>
+                }
+            )},
+        {blocked,
+            hoconsc:mk(
+                boolean(),
+                #{
+                    desc => ?DESC(blocked),
+                    required => true,
+                    example => false
+                }
+            )}
+    ] ++ fields(gcp_device);
+fields(import_result) ->
+    [
+        {errors,
+            hoconsc:mk(
+                non_neg_integer(),
+                #{
+                    desc => ?DESC(imported_counter_errors),
+                    required => true,
+                    example => 0
+                }
+            )},
+        {imported,
+            hoconsc:mk(
+                non_neg_integer(),
+                #{
+                    desc => ?DESC(imported_counter),
+                    required => true,
+                    example => 14
+                }
+            )}
+    ];
+fields(key) ->
+    [
+        {key,
+            hoconsc:mk(
+                binary(),
+                #{
+                    desc => ?DESC(key),
+                    required => true,
+                    example => <<
+                        "-----BEGIN CERTIFICATE-----\n"
+                        "MIIBEjCBuAIJAPKVZoroXatKMAoGCCqGSM49BAMCMBExDzANBgNVBAMMBnVudXNl\n"
```

Comment:
I am incredulous that is this example suitable, maybe we could let it empty, otherwise some vulnerability scanners may report private key leakage.

Reply:
This is available for GCP device CT ([c2_ec_cert](https://github.com/emqx/emqx/pull/11367/files#diff-1b5c04fcb1af507e8f002b11fc819a57838a54ec161dd72c798dd314ff19fe6f)), so I thought it would not be a problem to expose it here. I'll remove it anyway.